zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java

```diff
@@ -52,13 +57,21 @@ public FormEntity createNewEntity(final String id) {
   @Override
   public void updateEntity(final Record<Form> record, final FormEntity entity) {
     final Form value = record.getValue();
+    final var isDeleted = record.getIntent().equals(FormIntent.DELETED);
+
     entity
         .setVersion((long) value.getVersion())
         .setKey(value.getFormKey())
         .setBpmnId(value.getFormId())
         .setSchema(new String(value.getResource(), StandardCharsets.UTF_8))
         .setTenantId(value.getTenantId())
-        .setIsDeleted(record.getIntent().name().equals(FormIntent.DELETED.name()));
+        .setIsDeleted(isDeleted);
+
+    if (!isDeleted) {
+      // update local cache so that the form info is available immediately
+      final var cachedFormEntity = new CachedFormEntity(entity.getBpmnId(), entity.getVersion());
+      formCache.put(entity.getId(), cachedFormEntity);
```

Comment:
❓ should we remove from cache in case of `deleted`?

Reply:
Good point. I will add this!